sympy/parsing/mathematica.py

```diff
@@ -42,6 +46,10 @@ def parse(s):
         (r"\A(-? *[\d\.]+)([a-zA-Z].*)\Z",  # Implied multiplication - 2a
         lambda m: parse(m.group(1)) + "*" + parse(m.group(2))),
 
+        # Multiplication between any object and a function
+        (r"\A(.+)([*])(\w+\[[^\]]+[^\[]*\])\Z",
```

Comment:
Implemented it.

I [implemented](https://github.com/sympy/sympy/pull/12528/files#diff-5bead3fe0fbe39640e6bbc0055461aefR17) arithmetic operator for containing spaces( such as `a + b`) before implementing space multiplication so these expressions get matched before the parser matches for space multiplication.